frontend/src/app/team-management/search-team-management/search-team-management.component.ts

```diff
@@ -0,0 +1,147 @@
+import { Component } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { UserService } from '../../services/user.service';
+import { TeamService } from '../../services/team.service';
+import { Team } from '../../shared/types/model/Team';
+import { User } from '../../shared/types/model/User';
+import { BehaviorSubject, combineLatest, debounceTime, distinctUntilChanged, map } from 'rxjs';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { getRouteToTeam, getRouteToUserDetails } from '../../shared/routeUtils';
+import { Router } from '@angular/router';
+
+export interface FilteredUser extends User {
+  displayValue: string;
+  htmlValue: string;
+}
+
+export interface FilteredTeam extends Team {
+  displayValue: string;
+  htmlValue: string;
+}
+
+@Component({
+  selector: 'app-search-team-management',
+  templateUrl: './search-team-management.component.html',
+  styleUrl: './search-team-management.component.scss',
+})
+export class SearchTeamManagementComponent {
+  static MAX_SUGGESTIONS = 3;
+  search = new FormControl('');
+
+  filteredUsers$ = new BehaviorSubject<FilteredUser[]>([]);
+  filteredTeams$ = new BehaviorSubject<FilteredTeam[]>([]);
+  searchValue$ = new BehaviorSubject<string>('');
+
+  private teams: Team[] = [];
+  private users: User[] = [];
+
+  constructor(
+    private readonly userService: UserService,
+    private readonly teamService: TeamService,
+    private readonly router: Router,
+  ) {
+    combineLatest([teamService.getAllTeams(), userService.getUsers()])
+      .pipe(takeUntilDestroyed())
+      .subscribe(([teams, users]) => {
+        this.updateTeamsAndUsers(teams, users);
+        this.applyFilter(this.searchValue$.getValue());
+      });
+
+    this.search.valueChanges
+      .pipe(
+        takeUntilDestroyed(),
+        debounceTime(200),
+        map((v) => (v ? v.trim() : '')),
+        distinctUntilChanged(),
+      )
+      .subscribe((searchValue) => {
+        this.searchValue$.next(searchValue);
+      });
+
+    this.searchValue$.pipe(takeUntilDestroyed()).subscribe(() => {
+      this.applyFilter(this.searchValue$.getValue());
+    });
+  }
+
+  selectUser(user: User) {
+    this.search.setValue('');
+    this.router.navigateByUrl(getRouteToUserDetails(user.id)).then();
```

Comment:
Es wäre gut, wenn hier noch die Teamid mitgegeben würde. Dann wird der Member unter dem richtigen Team geöffnet.

Reply:
Habe ich angepasst, ich versuche anhand der zugewiesenen teams eine passende id zu finden. Falls nichts wird auf die "Basis-Seite" umgeleitet:
https://github.com/puzzle/okr/pull/758/commits/c442495634f1c7ba7e0ff7c9218cf8ffbff45656

Reply:
Ok, aber 1. Prio sollte das aktuelle Team haben, das offen ist,  wenn man auch 
teammitglied ist.

Reply:
Wurde gefixt in: https://github.com/puzzle/okr/pull/758/commits/f12c5f2d5354cc0f37b83ade682b4e1a1d6e2e40

Es machte aber keinen Sinn, so wie ich das beschrieben habe. Da es einem verwirrt hat, wenn man auf der Root page ist und dann auf ein willkürliches Team umgeleitet wird.

Das Verhalten nun so, dass die Seite nicht gewechselt wird, man bleibt auf der aktuellen Seite. Egal ob man auf "Alle Teams" oder einem Team ist.